torch/_torch_docs.py

```diff
@@ -4625,6 +4624,57 @@ def merge_dicts(*dicts):
     torch.return_types.median(values=tensor([-0.3982,  0.2270,  0.2488,  0.4742]), indices=tensor([1, 4, 4, 3]))
 """.format(**single_dim_common))
 
+add_docstr(torch.nanmedian,
+           r"""
+nanmedian(input) -> Tensor
+
+Computes the median of the values in :attr:`input`, ignoring ``NaN``s.
+
+This function is identical to :func:`torch.median` when there are no ``NaN`` values in :attr:`input`.
+When :attr:`input` has one or more ``NaN`` values, :func:`torch.median` will always return ``NaN``,
+while this function will return the median of the non-``NaN`` elements in :attr:`input`.
+If all the elements in :attr:`input` are ``NaN`` it will also return ``NaN``.
+
+Args:
+    {input}
+
+Example::
+
+    >>> a = torch.tensor([1, float('nan'), 3, 2])
+    >>> a.median()
+    tensor(nan)
+    >>> a.nanmedian()
+    tensor(2.)
+
+.. function:: nanmedian(input, dim=-1, keepdim=False, *, out=None) -> (Tensor, LongTensor)
+
+Computes the median of the values, and its index, along dimension :attr:`dim` of :attr:`input`, ignoring ``NaN``s.
+
+This function is identical to :func:`torch.median` when there are no ``NaN`` values in a reduced row. When a reduced row has
+one or more ``NaN`` values, :func:`torch.median` will always reduce it to ``NaN``, while this function will reduce it to the
+median of the non-``NaN`` elements. If all the elements in a reduced row are ``NaN`` then it will be reduced to ``NaN``, too.
+
+Args:
+    {input}
+    {dim}
+    {keepdim}
+
+Keyword args:
+    out ((Tensor, LongTensor), optional): The first tensor will be populated with the median values and the second tensor
```

Comment:
LongTensor is deprecated. You could say something like, "and the second tensor, which must have dtype long, ..."

Reply:
What about LongTensor in the return type?

Reply:
Same issue there. One problem with the use of the term LongTensor, for example, is that historically LongTensor meant a CPU tensor, because there was also CUDALongTensor.

Reply:
Maybe one thing for the doc hackathon is to remove all mentions of LongTensor, FloatTensor, etc from the docs. This might require adding a new returns section to the documents to describe the type of the returned tensors.